File: content/examples/other/leetcode/problems/valid-parentheses/example-1/solution.py
def is_valid_parentheses_stack(s):
    stack = []
    open_to_close = {"(": ")", "[": "]", "{": "}"}
    for ch in s:
        if ch in open_to_close:
            stack.append(ch)
            continue
        if not stack:
            return False
        open_ch = stack.pop()
        if open_to_close[open_ch] != ch:
            return False
    return len(stack) == 0


if __name__ == "__main__":
    print(is_valid_parentheses_stack("()[]{}"))
    print(is_valid_parentheses_stack("(]"))

File: content/examples/other/leetcode/problems/valid-parentheses/example-1/solution.js
function isValidParenthesesStack(s) {
  const stack = [];
  const openToClose = new Map([
    ["(", ")"],
    ["[", "]"],
    ["{", "}"],
  ]);

  for (const ch of s) {
    if (openToClose.has(ch)) {
      stack.push(ch);
      continue;
    }
    const open = stack.pop();
    if (open == null) return false;
    if (openToClose.get(open) !== ch) return false;
  }

  return stack.length === 0;
}

if (require.main === module) {
  console.log(isValidParenthesesStack("()[]{}"));
  console.log(isValidParenthesesStack("(]"));
}

module.exports = { isValidParenthesesStack };
